### DISCUSSION TITLE
Can I edit the main.workflow github-actions file locally?

### DISCUSSION THREAD

**Discussion by baruchiro** [community/community#25337] [Actions]
<p><a href="https://stackoverflow.com/questions/56600686/can-i-edit-the-main-workflow-github-actions-file-locally" rel="nofollow noopener">Asked in StackOverflow</a>.</p>
<p>If I change my main.workflow file locally, not in the master branch, the commit and push the change, I get this error from push command:</p>
<pre><code>&gt; git diff
diff --git a/.github/main.workflow b/.github/main.workflow
index 135d8ea..0a13a28 100644
--- a/.github/main.workflow
+++ b/.github/main.workflow
@@ -6,7 +6,7 @@ workflow "Build and Test in PR" {
   ]
 }

-action ".NET Core CLI" {
+action ".NET Core CLI" {
   uses = "./.github/net-core/"
   args = "test"
 }

&gt; git push
! [remote rejected] my-branch -&gt; my-branch (refusing to allow an integration to create or update .github/main.workflow)
error: failed to push some refs to 'https://github.com/my-user-name/my-repo.git'
</code></pre>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>HI <a class="mention" href="/baruchiro">@baruchiro</a>,</p>
<p>Thanks for being here! If you are using GitHub desktop there is currently a bug that causes this error when attempting to push commits from GitHub Actions. As a workaround, you can use the <a href="https://developer.github.com/v3/git/" rel="nofollow noopener">Git API</a> until the fix for this issue is released.</p>
<p>Since the issue used for tracking this bug is not external to the public, I would recommend checking in with us again in a week or two for an update.</p>

**Comment by baruchiro** [comment]
<p>I’m afraid I did not quite understand.<br>
As you can see, I did not use <strong>Github Desktop</strong> but <strong>Git CLI</strong>.</p>

**Comment by wei** [comment]
<p>Hi <a class="mention" href="/baruchiro">@baruchiro</a> ,</p>
<p>You’re probably using a Git client of some sort which is using an OAuth token to pull/push to Github. I would suggest updating upstream to use ssh or https with your own credentials / personal access tokens instead of OAuth tokens given to third-party apps.</p>

**Comment by lee-dohm** [comment]
<p>Yes, the main.workflow can be edited locally and changes pushed to the GitHub-hosted repository. The error message doesn’t mean exactly what you’re describing though:</p>
<pre><code>refusing to allow an integration to create or update .github/main.workflow
</code></pre>
<p>An “integration” is a GitHub App, typically. Additionally, the <code>GITHUB_TOKEN</code> that is supplied to Actions is a GitHub App token, so if you were using an Action to modify <code>main.workflow</code> then this would be one reason you’re getting the error you show.</p>
<p>But as a real user, not a GitHub App or Action, you modified <code>main.workflow</code> locally and have write access to the repository and branch you’re pushing to, yes, you can modify <code>main.workflow</code> locally to your heart’s content :grinning:</p>
<p>I hope that helps!</p>

**Comment by case** [comment]
<p>Hi <span class="mention">@AndreaG</span>,</p>
<p>&gt; Since the issue used for tracking this bug is not external to the public, I would recommend checking in with us again in a week or two for an update.</p>
<p>I’m running into this issue, and curious if there’s any news on the fix from GitHub’s side?</p>
<p>Thanks,</p>
<p>-Eric</p>

**Comment by Tyrrrz** [comment]
<p>I’m observing this through git command line. Marking this issue as solved is very misleading.</p>

**Comment by hXtreme** [comment]
<p>Thanks for the explanation!</p>
<p>Is there a way to allow permission for an ‘Integration’ (in my case a restyle bot) permission to modify workflow?</p>
<p>Log: <a href="https://restyled.io/gh/hXtreme/repos/HCP-Project/jobs/140246" rel="nofollow noopener">https://restyled.io/gh/hXtreme/repos/HCP-Project/jobs/140246</a></p>

**Comment by Santiago8888** [comment]
<p>For the interested, I got the error trying to push from VS, but not from the terminal, (I am using signed commits). OS: Ubuntu.</p>
